R-package/R/create-release-files.R

```diff
@@ -0,0 +1,411 @@
+          }
+        } else{
+          if (partID %in% parts_sheet[["partID"]]) {
+            if (partID %in% names(dictionary)) {
+              validated_file <- TRUE
+            } else{
+              logger::log_warn(paste0(single_part,
+                                      " does not have a matching sheet."))
+              errors <- TRUE
+            }
+            
+          } else {
+            logger::log_warn(paste0(partID,
+                                    " is not found in parts sheet."))
+            errors <- TRUE
+          }
+        }
+      } else if (file_type == files$file_type$categories$csv) {
+        if (nrow(set_info) >= 1) {
+          logger::log_warn(paste0(partID,
+                                  " is recorded for csv but is found in sets."))
+          errors <- TRUE
+        } else{
+          if (partID %in% parts_sheet[["partID"]]) {
+            if (partID %in% names(dictionary)) {
+              validated_file <- TRUE
+            } else{
+              logger::log_warn(paste0(single_part,
+                                      " does not have a matching sheet."))
+              errors <- TRUE
+            }
+            
+          } else {
+            logger::log_warn(paste0(partID,
+                                    " is not found in parts sheet."))
+            errors <- TRUE
+          }
+          
+        }
+      } else{
+        logger::log_warn(paste0(partID,
+                                " has an unrecognized fileType of ",
+                                file_type))
+        errors <- TRUE
+      }
+      
+      # Append valid files list
+      if (validated_file) {
+        files_to_extract[[fileID]] <- list(
+          file_name = file_name,
+          file_type = file_type,
+          partID = partID,
+          add_headers = add_headers,
+          destination = destination,
+          osf_location = osf_location,
+          github_location = github_location
+        )
+      }
+    }
+    # Will move with development
+    # Will become stop once function development is finished
+    if (errors != "") {
+      
+      warning("Errors were detected further building cannot continue please check the log for additional info")
+    }
+    return(files_to_extract)
+  }
+
+#' Create Files
+#'
+#' Function responsible for creating the release files based on output from validate_files_sheet
+#'
+#' @param files_to_extract List output from validate_files_sheet
+#' @param dictionary openxlsx environment object storing the dictionary
+create_files <-
+  function(files_to_extract,
+           dictionary) {
+    # Loop over files to extract based on fileID
+    for (fileID in names(files_to_extract)) {
+      current_file_info <- files_to_extract[[fileID]]
+      
+      # Create a write_directory based on destination and saving location
+      write_dir <- ""
+      if (current_file_info$destination == "github") {
+        write_dir <- file.path(
+          getwd(),
```

Comment:
The path `getwd() + odm_dictionary$tmp_dictionary_directory` can be put in a variable and reused.